src/main/java/util/ApiQuotaManager.java

```diff
@@ -0,0 +1,180 @@
+public class ApiQuotaManager {
+
+    private static final Logger logger = HTLog.get(ApiQuotaManager.class);
+
+    /**
+     * The amount that is set aside for manual operations by the user.
+     */
+    private static final int API_QUOTA_BUFFER = 200;
+
+    /**
+     * The default refresh period.
+     */
+    private static final int DEFAULT_REFRESH_PERIOD_IN_MINS = 1;
+
+    /**
+     * The uninitialized value for uninitialized variables.
+     */
+    private static final int UNINITIALIZED = -1;
+
+    /**
+     * The amount of the available remaining API quota.
+     */
+    private int remainingApiQuota = UNINITIALIZED;
+
+    /**
+     * The number of API calls used in most recent refresh.
+     */
+    private int apiCallsUsedInRecentRefresh = UNINITIALIZED;
+
+    /**
+     * The duration between each refresh of the data store.
+     */
+    private long refreshDurationInMinutes = DEFAULT_REFRESH_PERIOD_IN_MINS;
+
+    private final TickingTimer refreshTimer;
+
+    public ApiQuotaManager(Logic logic) {
+        this.refreshTimer = new TickingTimer("Refresh Timer",
+                (int) Utility.minsToSecs(DEFAULT_REFRESH_PERIOD_IN_MINS),
+                UI.status::updateTimeToRefresh, logic::refresh, TimeUnit.SECONDS);
+        this.refreshTimer.start();
+        UI.events.registerEvent((RefreshTimerTriggeredEventHandler) this::updateSyncRefreshRate);
+    }
+
+    /**
+     * @return The duration between each refresh of the data store.
+     */
+    public long getRefreshDurationInMinutes() {
+        return refreshDurationInMinutes;
+    }
+
+    /**
+     * Restarts the refreshTimer.
+     */
+    public void restartRefreshTimer(){
+        this.refreshTimer.restart();
+    }
+
+    /**
+     * Updates the period of the refreshTimer for synchronization of the data store.
+     * The initial call to this method does not update the period of the refreshTimer.
+     * @param e RefreshTimerTriggeredEvent object that holds the current API quota information.
+     */
+    private void updateSyncRefreshRate(RefreshTimerTriggeredEvent e) {
+        updateSyncRefreshRate(e.getApiQuotaInfo().getRemainingQuota(),
+                              e.getApiQuotaInfo().getNextRefreshInMinutesFromNow());
+    }
+
+    private void updateSyncRefreshRate(int newRemainingApiQuota, long nextRefreshInMinuteFromNow) {
+
+        if (remainingApiQuota == UNINITIALIZED){
+            remainingApiQuota = newRemainingApiQuota;
+            return;
+        }
+
+        // apiCallsUsedInRecentRefresh will not get initialised in the first attempt if an API quota top up has occurred
+        // since the recent refresh.
+        if (apiCallsUsedInRecentRefresh == UNINITIALIZED){
+            if (remainingApiQuota >= newRemainingApiQuota){
+                apiCallsUsedInRecentRefresh = remainingApiQuota - newRemainingApiQuota;
+            } else {
+                remainingApiQuota = newRemainingApiQuota;
+                return;
+            }
+        }
```

Comment:
I have a feeling this entire block can be removed without affecting the final behavior. Try removing it and running the unit tests. If all tests pass, either the block is redundant or your tests are weak.